src/main.cpp

```diff
@@ -816,14 +823,58 @@ static bool run_dedicated_server(const GameParams &game_params, const Settings &
 	if (cmd_args.exists("migrate"))
 		return migrate_database(game_params, cmd_args);
 
-	// Create server
-	Server server(game_params.world_path, game_params.game_spec, false,
-		bind_addr.isIPv6());
-	server.start(bind_addr);
+	if (cmd_args.exists("terminal")) {
+#if USE_CURSES
```

Comment:
Maybe the contents of this could be separated into a new interactive_init() function?